include/nbl/builtin/hlsl/cpp_compat/intrinsics.hlsl

```diff
@@ -161,7 +164,17 @@ inline FloatingPoint isnan(NBL_CONST_REF_ARG(FloatingPoint) val)
 #endif
 }
 
-template<typename FloatingPoint>
+template <typename Integer NBL_FUNC_REQUIRES(hlsl::is_integral_v<Integer>)
+inline bool isnan(Integer val)
+{
+	using AsUint = typename unsigned_integer_of_size<sizeof(Integer)>::type;
+	using AsFloat = typename float_of_size<sizeof(Integer)>::type;
+
+	AsUint asUint = bit_cast<AsUint, Integer>(val);
+	return bool((ieee754::extractBiasedExponent<Integer>(val) == ieee754::traits<AsFloat>::specialValueExp) && (asUint & ieee754::traits<AsFloat>::mantissaMask));
+}
```

Comment:
integer versions of isnan and isinf really shouldn't exist anywhere publicly even if C++23 or C++26 adds some weird version of them

Reply:
the `bit_cast` should happen in the floating point isnan if this function got made to work around the `-ffast-math` (the ffast-math workaround relies on you not  calling `std::` versions of the functions)